### DISCUSSION TITLE
Hackathon: 2024-08-20

### DISCUSSION THREAD

**Discussion by mfisher87** [earthaccess-dev/earthaccess#791] [Hack days]
Reporting out on earthaccess hack days.

Please create a new comment (at the very bottom, use the text box with a button that says "comment", not "reply") to write about what you did today!

Please include links to GitHub issues or pull requests relevant to your work, and use those issues or pull requests as the "source of truth" for documenting the work; this makes future readers jobs easier :) It's OK to duplicate some information while writing a summary here!

Use the "reply" feature to have a discussion under any given comment.

**Comment by mfisher87** [comment]
@Sherwin-14 @DeanHenze @danielfromearth  and I continued to make progress on the integration testing branch #749 . Danny updated our collection generation script to work for all providers that earthaccess supports today (17), with comments describing how to scale out to all providers (https://cmr.earthdata.nasa.gov/ingest/providers). I have work in progress to update the onprem download integration tests to use those source files. I'm planning to do a bit more and push soon.

@DeanHenze and I discussed a possible future top-level API function `earthaccess.search_virtual_dataset()`.

**Reply by chuckwondo** [reply]
> @DeanHenze and I discussed a possible future top-level API function `earthaccess.search_virtual_dataset()`.

Perhaps `find_virtual_dataset`?

**Reply by mfisher87** [reply]
Good call! Related: https://github.com/nsidc/earthaccess/issues/770
